openwisp_controller/config/whois/service.py

```diff
@@ -108,11 +113,19 @@ def trigger_whois_lookup(self):
         Trigger WHOIS lookup based on the conditions of `_need_whois_lookup`.
         Task is triggered on commit to ensure redundant data is not created.
         """
-        if self._need_whois_lookup(self.device.last_ip):
+
+        fetch_whois, whois_info_exists = self._need_whois_lookup(self.device.last_ip)
+        if fetch_whois:
             transaction.on_commit(
                 lambda: fetch_whois_details.delay(
                     device_pk=self.device.pk,
                     initial_ip_address=self.device._initial_last_ip,
                     new_ip_address=self.device.last_ip,
                 )
             )
+        elif whois_info_exists and self.is_whois_enabled:
```

Comment:
WHOIS information should exist only when WHOIS is enabled. Do you mean that if WHOIS is disabled after checksum occurs and WHOIS info is added in DB? What are the consequences if we show fuzzy location for that last_ip only whether WHOIS is enabled or not?